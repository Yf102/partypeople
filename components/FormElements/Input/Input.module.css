.text {
    background: red;
    color: #fff;
    border-radius: 0 0 5px 5px;
    display: flex;
    align-items: center;
    justify-content: center;
    font-size: 14px;
    overflow: hidden;
}