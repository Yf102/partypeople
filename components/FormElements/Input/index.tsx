import React from 'react'
import { FieldError } from 'react-hook-form'
import styles from './Input.module.css'

type InputProps = React.DetailedHTMLProps<
  React.InputHTMLAttributes<HTMLInputElement>,
  HTMLInputElement
>

type InputCustomProps = {
  label?: string
  error?: FieldError
}

const Input = React.forwardRef<HTMLInputElement, InputProps & InputCustomProps>(
  (props, ref) => {
    const { label, error } = props

    return (
      <div className='w-full pt-0 mb-3'>
        {label && <label data-testid='input-label-element'>{label}</label>}
        <input
          ref={ref}
          {...props}
          className='relative h-12 w-full px-4 py-1 bg-white border-0 rounded-md outline-none focus:outline-none text-center text-xl text-black'
        />
        {error && <div className={styles.text}>{error.message}</div>}
      </div>
    )
  }
)

Input.displayName = 'Input'

export default Input
