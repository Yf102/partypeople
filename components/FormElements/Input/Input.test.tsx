import React from 'react'
import { render, screen } from 'tests/test-utils'
import Input from 'components/FormElements/Input/index'

describe('Input', () => {
  it('It renders Input with label', async () => {
    const { asFragment } = render(
      <Input label='Testing Label' placeholder='23.351723' />
    )
    expect(screen.getByTestId('input-label-element')).toBeTruthy()
    expect(asFragment()).toMatchSnapshot()
  })

  it('It renders Input with NO label', async () => {
    const { asFragment } = render(<Input placeholder='23.351723' />)

    expect(screen.queryByTestId('input-label-element')).not.toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })
})
