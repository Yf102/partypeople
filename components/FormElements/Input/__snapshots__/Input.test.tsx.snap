// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input It renders Input with NO label 1`] = `
<DocumentFragment>
  <div
    class="w-full pt-0 mb-3"
  >
    <input
      class="relative h-12 w-full px-4 py-1 bg-white border-0 rounded-md outline-none focus:outline-none text-center text-xl text-black"
      placeholder="23.351723"
    />
  </div>
</DocumentFragment>
`;

exports[`Input It renders Input with label 1`] = `
<DocumentFragment>
  <div
    class="w-full pt-0 mb-3"
  >
    <label
      data-testid="input-label-element"
    >
      Testing Label
    </label>
    <input
      class="relative h-12 w-full px-4 py-1 bg-white border-0 rounded-md outline-none focus:outline-none text-center text-xl text-black"
      label="Testing Label"
      placeholder="23.351723"
    />
  </div>
</DocumentFragment>
`;
